src/dac.rs

```diff
@@ -0,0 +1,208 @@
+    Mamp8 = 0b1000,
+    /// Unmask bits[9:0] of LFSR/ triangle amplitude equal to 1023 
+    Mamp9 = 0b1001,
+    /// Unmask bits[10:0] of LFSR/ triangle amplitude equal to 2047
+    Mamp10 = 0b1010, 
+    // Unmask bits[11:0] of LFSR/ triangle amplitude equal to 4095
+    Mamp11 = 0b1011, 
+}
+
+/// Represents a Digital to Analog Converter (DAC) peripheral.
+pub struct Dac {
+    regs: DAC1,
+    bits: DacBitAlignment,
+}
+
+// todo: Calculate the VDDA vref, as you do with onboard ADCs!
+impl Dac 
+{
+    /// Initialize a DAC peripheral, including  enabling and resetting
+
+    pub fn new(regs: DAC1, bits: DacBitAlignment) -> Self {
+        
+        let rcc = unsafe { &(*RCC::ptr()) };
+
+        rcc.apb1enr.modify(|_,  w| w.dac1en().set_bit());
+        rcc.apb1rstr.modify(|_, w| w.dac1rst().set_bit());
+        rcc.apb1rstr.modify(|_, w| w.dac1rst().clear_bit());
+
+        Self {
+            regs,
+            bits,
+        }
+    }
+
+    pub fn enable_channel(&mut self, channel: DacChannel) {
+        let cr = &self.regs.cr;
+
+        cr.modify(|_, w| match channel {
+            DacChannel::One => w.en1().set_bit(),
+            DacChannel::Two => w.en2().set_bit(),
+        });
+    }
+
+    pub fn set_trigger(&mut self, channel: DacChannel, trigger: Trigger) {
+        let cr = &self.regs.cr;
+
+        match channel {
+            DacChannel::One => {
+                cr.modify(|_, w| unsafe {
+                    w.ten1().set_bit();
+                    w.tsel1().bits(trigger as u8)
+                });
+            }
+            DacChannel::Two => {
+                cr.modify(|_, w| unsafe {
+                    w.ten2().set_bit();
+                    w.tsel2().bits(trigger as u8)
+                });
+            }
+        }
+    }
+
+    pub fn enable_noise_gen(&mut self, channel: DacChannel, mamp: Mamp) {
+        let cr = &self.regs.cr;
+
+        match channel {
+            DacChannel::One => {
+                cr.modify(|_, w| unsafe {
+                    w.wave1().noise();
+                    w.mamp1().bits(mamp as u8)
+                });
+            }
+            DacChannel::Two => {
+                cr.modify(|_, w| unsafe {
+                    w.wave2().noise();
+                    w.mamp2().bits(mamp as u8)
+                });
+            }
+        }
+
+    }
+
+    pub fn enable_triangle_gen(&mut self, channel: DacChannel, mamp: Mamp) {
+        let cr = &self.regs.cr;
+
+        match channel {
+            DacChannel::One => {
+                cr.modify(|_, w| unsafe {
+                    w.wave1().triangle();
+                    w.mamp1().bits(mamp as u8)
+                });
+            }
+            DacChannel::Two => {
+                cr.modify(|_, w| unsafe {
+                    w.wave2().triangle();
+                    w.mamp2().bits(mamp as u8)
+                });
+            }
+        }
+
+    }
+
+    pub fn write_data(&mut self, data: u32) {
```

Comment:
the use of the `data` argument looks strange because *all* the 32 bits are written into one the registers but the registers have several number of reserved bits. e.g. one should only write 8 bits of data into (the lower byte of) DHR8R1.

Reply:
u32 is a requirement here

```
#[doc = "channel1 8-bit right aligned data holding register\n\nThis register you can [`read`](crate::generic::Reg::read), [`reset`](crate::generic::Reg::reset), [`write`](crate::generic::Reg::write), [`write_with_zero`](crate::generic::Reg::write_with_zero), [`modify`](crate::generic::Reg::modify). See [API](https://docs.rs/svd2rust/#read--modify--write-api).\n\nFor information about available fields see [dhr8r1](dhr8r1) module"]
pub type DHR8R1 = crate::Reg<u32, _DHR8R1>;
```